ipatests/pytest_ipa/integration/tasks.py

```diff
@@ -2533,6 +2534,27 @@ def get_healthcheck_version(host):
     return healthcheck_version
 
 
+def wait_for_ipa_to_start(host, timeout=60):
+    """Wait up to timeout seconds for ipa to start on a given host.
+
+    If DS is restarted, and SSSD must be online, please consider using
+    wait_for_sssd_domain_status_online(host) in the test after calling
+    this method.
+    """
+    pattern = 'STOPPED'
+    interval = 1
+    end_time = datetime.now() + timedelta(seconds=timeout)
+    for _i in range(0, timeout, interval):
+        if datetime.now() > end_time:
+            raise RuntimeError("Request timed out")
+        time.sleep(interval)
+        result = host.run_command(
+            [paths.IPACTL, "status"]
```

Comment:
The `ipactl status` command has been fixed recently (https://github.com/freeipa/freeipa/commit/928ab51be669d7af7a28205acb9eb7e9b46e189e) and now returns 0 only when all services are running. This means that you can simplify this method and only need to check the exit code to ensure that IPA is up, no need for output string parsing.

Reply:
That's right. Fixed.